smartsim/settings/arguments/launch/alps.py

```diff
@@ -29,21 +29,30 @@
 import typing as t
 
 from smartsim.log import get_logger
+from smartsim.settings.dispatch import ShellLauncher, dispatch, make_shell_format_fn
 
-from ...common import StringArgument, set_check_input
+from ...common import set_check_input
 from ...launchCommand import LauncherType
-from ..launchArgBuilder import LaunchArgBuilder
+from ..launchArguments import LaunchArguments
 
 logger = get_logger(__name__)
+_as_aprun_command = make_shell_format_fn(run_command="aprun")
 
 
-class AprunArgBuilder(LaunchArgBuilder):
+@dispatch(with_format=_as_aprun_command, to_launcher=ShellLauncher)
```

Comment:
I'd be willing to add a docstring where we define `disapatch = DEFAULT_DISPATCHER.dispatch`, but if at all possible I'd like to avoid documenting every usage of `dispatch` as a decorator (just in case we change how it works in the future, it'd be a lot of touch points we would need to remember to update).

Will add a docstring for `disptach` in the `dispatch` module!